client/src/renderer/components/Editor/index.jsx

```diff
@@ -38,15 +41,35 @@ export default class Editor extends Component {
     this.contentEditableDiv.current.innerHTML = "";
   }
 
+  //returns absolute values of caret's start/end positions
+  getCaretPos() {
+    const editor = document.getElementById("editor");
+    var positions = { pos1: 0, pos2: 0 };
+
+    var range = document.getSelection().getRangeAt(0);
+    var preSelectionRange = range.cloneRange();
+    preSelectionRange.selectNodeContents(editor);
+    preSelectionRange.setEnd(range.startContainer, range.startOffset);
+    positions.pos1 = preSelectionRange.toString().length;
+    positions.pos2 = positions.pos1 + range.toString().length;
+    console.log("Editor: ", positions.pos1, positions.pos2);
+    return positions;
+  }
+
   render(props) {
+    const dispatch = useDispatch();
+    const caretPos = useSelector(state => state.caret);
+
     return (
       <div>
         <ToolBar />
 
         <div
           ref={this.contentEditableDiv}
           class="editor"
+          id="editor"
           contenteditable="true"
+          onclick={() => dispatch(updateCaretPos(this.getCaretPos()))}
```

Comment:
Please use camel case.
```suggestion
          onClick={() => dispatch(updateCaretPos(this.getCaretPos()))}
```

Reply:
Also, please use an event handler rather than an inline function. Makes it easier to append functionality in future, e.g.
```suggestion
          onclick={this.handleDocumentClick}
```

Then define `handleDocumentClick` above the render function.

Reply:
this.fixed(this)